File: sdks/typescript/examples/basic-chat.ts
import { KernelClient } from '../src';

async function main() {
  console.log('Connecting to claw-kernel...');
  const client = await KernelClient.connect();

  const info = await client.info();
  console.log(`Connected! Kernel version: ${info.version}, protocol: v${info.protocolVersion}`);

  const session = await client.createSession({
    systemPrompt: 'You are a helpful assistant. Be concise.',
    maxTurns: 10,
  });

  console.log(`Session created: ${session.id}`);
  console.log('Sending message...\n');

  const response = await session.send('What is 2 + 2?');
  console.log('Response:', response);

  await session.close();
  client.close();
  console.log('\nDone!');
}

main().catch(console.error);

File: sdk/python/examples/basic_chat.py
"""
Basic chat example using the synchronous KernelClient.

Run with: python basic_chat.py
"""

from claw_kernel import KernelClient, SessionConfig


def main():
    with KernelClient() as client:
        # Show kernel info
        info = client.info()
        print(f"Connected to claw-kernel v{info.version}")
        print(f"Active provider: {info.active_provider} / {info.active_model}")
        print()

        # Create a session
        session_id = client.session.create(
            SessionConfig(system_prompt="You are a friendly assistant.")
        )
        print(f"Session created: {session_id}")

        # Send messages in a loop
        while True:
            user_input = input("\nYou: ").strip()
            if user_input.lower() in ("exit", "quit", "q"):
                break

            print("Assistant: ", end="", flush=True)
            for token in client.session.send(session_id, user_input):
                print(token, end="", flush=True)
            print()

        # Clean up
        client.session.destroy(session_id)
        print("\nSession destroyed. Goodbye!")


if __name__ == "__main__":
    main()
